cmd/publicize/main.go

```diff
@@ -0,0 +1,189 @@
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"k8s.io/test-infra/prow/githubeventserver"
+	"k8s.io/test-infra/prow/interrupts"
+	"k8s.io/test-infra/prow/logrusutil"
+
+	"k8s.io/test-infra/pkg/flagutil"
+	"k8s.io/test-infra/prow/config/secret"
+	prowflagutil "k8s.io/test-infra/prow/flagutil"
+)
+
+type Config struct {
+	Bonds []Bond `json:"bonds,omitempty"`
+}
+
+func (c *Config) validate() error {
+	var errs []error
+	for index, bond := range c.Bonds {
+		if err := bond.validate(); err != nil {
+			errs = append(errs, fmt.Errorf("bond[%d]: %w", index, err))
+		}
+	}
+
+	return kerrors.NewAggregate(errs)
+}
+
+// bond holds the information of which repos are bonded to each other.
+type Bond struct {
+	From string `json:"from"`
+	To   string `json:"to"`
+}
+
+func (b *Bond) validate() error {
+	for _, value := range []string{b.From, b.To} {
+		if len(strings.Split(value, "/")) != 2 {
+			return fmt.Errorf("%s should be in org/repo format", value)
+		}
+	}
+	return nil
+}
+
+type options struct {
+	configPath               string
+	gitName                  string
+	gitEmail                 string
+	bonds                    map[string]string
+	dryRun                   bool
+	githubEventServerOptions githubeventserver.Options
+	github                   prowflagutil.GitHubOptions
+	git                      prowflagutil.GitOptions
+
+	webhookSecretFile string
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.BoolVar(&o.dryRun, "dry-run", true, "Dry run for testing. Uses API tokens but does not mutate.")
+
+	fs.StringVar(&o.configPath, "config-path", "/etc/config/config.yaml", "Path to publicize configuration.")
+	fs.StringVar(&o.webhookSecretFile, "hmac-secret-file", "/etc/webhook/hmac", "Path to the file containing the GitHub HMAC secret.")
+
+	fs.StringVar(&o.gitName, "git-name", "", "The name to use on the git commit. Requires --git-email. If not specified, uses the system default.")
+	fs.StringVar(&o.gitEmail, "git-email", "", "The email to use on the git commit. Requires --git-name. If not specified, uses the system default.")
+
+	for _, group := range []flagutil.OptionGroup{&o.github} {
+		group.AddFlags(fs)
+	}
+	o.githubEventServerOptions.Bind(fs)
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		logrus.WithError(err).Fatalf("cannot parse args: '%s'", os.Args[1:])
+	}
+	return o
+}
+
+func (o *options) Validate() error {
+	if (o.gitEmail == "") != (o.gitName == "") {
+		return errors.New("--git-name and --git-email must be specified")
+	}
+
+	for _, group := range []flagutil.OptionGroup{&o.github} {
+		if err := group.Validate(o.dryRun); err != nil {
+			return err
+		}
+	}
+
+	bytes, err := ioutil.ReadFile(o.configPath)
+	if err != nil {
+		return fmt.Errorf("Couldn't read publicize configuration file: %v", o.configPath)
+	}
+
+	var c Config
+	if err := yaml.Unmarshal(bytes, &c); err != nil {
+		return errors.New("Couldn't unmarshal publicize configuration")
```

Comment:
Why eat the error and not returning the actual one (which may contain somehting useful about hwy the unmarshal failed?